### PR TITLE
refactor (jkube-kit/common) : Add `launderThrowable` method to JKubeException

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

+ Port KubernetesClientException.launderThrowable to JKubeException so
      that we can wrap checked exceptions inside it.
+ Use abovementioned method in `ProfileUtil.blendProfileWithConfiguration`
+ Fix typo in method name `extractProcesssorConfiguration`

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] My code follows the style guidelines of this project
 - [X] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [X] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [X] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [X] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/jkubeio/ci)

Started new GH workflow run for https://github.com/eclipse/jkube/pull/2769 (_2024-03-05T14:18:35Z_)

:gear: [JKube E2E Tests (8157762513)](https://github.com/jkubeio/ci/actions/runs/8157762513) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#2769)](https://github.com/jkubeio/ci/actions/runs/8157762513/job/22298158007)

File: gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/KubernetesWatchTask.java [hunk 1]
```diff
@@ -88,11 +88,7 @@ private WatcherContext createWatcherContext() throws IOException {
   }
 
   private ProcessorConfig extractWatcherConfig() {
```
Comment (manusa) [inline gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/KubernetesWatchTask.java:90]: Is this method even necessary?

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/JKubeException.java [hunk 2]
```diff
@@ -19,4 +19,32 @@ public JKubeException(String message) {
...
+  private static RuntimeException processCause(Throwable cause) {
+    if (cause instanceof RuntimeException) {
+      return ((RuntimeException) cause);
+    } else if (cause instanceof Error) {
+      throw ((Error) cause);
+    } else if (cause instanceof InterruptedException) {
+      Thread.currentThread().interrupt();
+    }
+    return null;
+  }
```
Comment (manusa) [inline jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/JKubeException.java:49]: This method is not reused elsewhere, there's no point in extracting it and then doing a redundant if-else check in launderThrowable

File: gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java [hunk 3]
```diff
@@ -193,11 +193,7 @@ protected final List<File> resolveResourceSourceDirectory() {
 
 
   protected ProcessorConfig extractGeneratorConfig() {
```
Comment (manusa) [inline gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java:195]: Doesn't seem to be overridden anywhere, no point in keeping a separate method.

File: kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/AbstractDockerMojo.java [hunk 4]
```diff
@@ -617,20 +617,12 @@ protected GeneratorContext.GeneratorContextBuilder generatorContextBuilder() thr
...
-        try {
-            return ProfileUtil.blendProfileWithConfiguration(ProfileUtil.GENERATOR_CONFIG, profile, ResourceUtil.getFinalResourceDirs(resourceDir, environment), generator);
-        } catch (IOException e) {
-            throw new IllegalArgumentException("Cannot extract generator config: " + e, e);
-        }
+        return ProfileUtil.blendProfileWithConfiguration(ProfileUtil.GENERATOR_CONFIG, profile, ResourceUtil.getFinalResourceDirs(resourceDir, environment), generator);
     }
 
     // Get enricher config
     protected ProcessorConfig extractEnricherConfig() {
```
Comment (manusa) [inline kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/AbstractDockerMojo.java:624]: Doesn't seem to be overridden anywhere, no point in keeping a separate method.

File: kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/develop/WatchMojo.java [hunk 5]
```diff
@@ -143,11 +143,7 @@ protected GeneratorContext.GeneratorContextBuilder generatorContextBuilder() thr
 
     // Get watcher config
     private ProcessorConfig extractWatcherConfig() {
```
Comment (manusa) [inline kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/develop/WatchMojo.java:145]: Is this method even necessary?

**Review by manusa — CHANGES_REQUESTED** [review]
Please check inlined comments

File: jkube-kit/profile/src/main/java/org/eclipse/jkube/kit/profile/ProfileUtil.java [hunk 6]
```diff
@@ -100,16 +101,19 @@ private static Profile inheritFromParentProfile(Profile aProfile, File resourceD
...
+        try {
+            // Get specified profile or the default profile
+            ProcessorConfig profileConfig = extractProcesssorConfiguration(configExtractor, profile, resourceDirs);
 
-        return ProcessorConfig.mergeProcessorConfigs(config, profileConfig);
+            return ProcessorConfig.mergeProcessorConfigs(config, profileConfig);
+        } catch (IOException ioException) {
+            throw JKubeException.launderThrowable("Error in extracting processor configuration", ioException);
+        }
     }
```
Comment (manusa) [inline jkube-kit/profile/src/main/java/org/eclipse/jkube/kit/profile/ProfileUtil.java:117]: If you carefully check the method, the IOExcpetion is only thrown at the `public static Profile findProfile(String profileArg, File... resourceDirs)` method.

Maybe this is the method where we want to throw the JKubeException instead.

**Review by manusa — CHANGES_REQUESTED** [review]
https://github.com/eclipse/jkube/pull/2769/files#r1513238396

File: jkube-kit/profile/src/main/java/org/eclipse/jkube/kit/profile/ProfileUtil.java [hunk 7]
```diff
@@ -100,16 +100,19 @@ private static Profile inheritFromParentProfile(Profile aProfile, File resourceD
...
+                                                                ProcessorConfig config) {
+        try {
+            // Get specified profile or the default profile
+            ProcessorConfig profileConfig = extractProcesssorConfiguration(configExtractor, profile, resourceDirs);
 
-        return ProcessorConfig.mergeProcessorConfigs(config, profileConfig);
+            return ProcessorConfig.mergeProcessorConfigs(config, profileConfig);
+        } catch (IOException ioException) {
+            throw JKubeException.launderThrowable("Error in extracting processor configuration", ioException);
+        }
```
Comment (manusa) [inline jkube-kit/profile/src/main/java/org/eclipse/jkube/kit/profile/ProfileUtil.java:115]: The point is not to have this. You can safely remove the throws IOException from the extractProcesssorConfiguration (sic) method.

Please, take your time and do things properly and thinking twice. It's the third iteration already for a really simple change.

Comment (rohanKanojia) [reply jkube-kit/profile/src/main/java/org/eclipse/jkube/kit/profile/ProfileUtil.java]: Sorry, I think it should be okay now

**Review by manusa — APPROVED** [review]

**Pull request merged**